drivers/sensor/icm42670S/icm42670S.c

```diff
@@ -494,9 +491,7 @@ static int icm42670S_attr_set(const struct device *dev,
 			LOG_ERR("Not supported ATTR");
 			return -EINVAL;
 		}
-		break;
-		
-	case SENSOR_CHAN_APEX_MOTION:
+	} else if ((enum sensor_channel_icm42670S)chan == SENSOR_CHAN_APEX_MOTION) {
```

Comment:
Why are you casting to enum in this if statement and not in previous ones?

Reply:
Because previous one are generic enum (enum sensor_channel) and this one is icm42670S specific defined in specific include/zephyr/drivers/sensor/icm42670S.h